File: isPalindromeNumber/code.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0 or (x % 10 == 0 and x != 0):
            return False
        reverse = 0
        while x > reverse:
            last_digit = x % 10
            reverse = reverse * 10 + last_digit
            x = x // 10

        return x == reverse or x == reverse // 10

File: isPalindromeNumber/code.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        // Handle edge cases
        if(x < 0 || (x != 0 && x % 10 == 0)) return false;
        
        int reversed_half = 0;
        while(x > reversed_half){
            int last_digit = x % 10;
            reversed_half = reversed_half * 10 + last_digit;
            x = x / 10;
        }
        
        return x == reversed_half || x == reversed_half/10;
    }
};
